### PR TITLE
Set current working directory for procedural macros

### LINKED ISSUE #11079: set current working directory for procedural macros
In 

https://github.com/rust-analyzer/rust-analyzer/blob/8dc3a270f6a893a592db1a5fe3cf8752521987e1/crates/proc_macro_srv/src/cli.rs#L8-L24

nothing sets currend dir explicitly, it gets inherited. This is wrong, as proc-macros generally can look at pwd. Note that this is per-macro thing -- we run a single expander process, but each crates' macros want to see their own pwd. 

I think we need to add `current_dir` to this struct: 

https://github.com/rust-analyzer/rust-analyzer/blob/8dc3a270f6a893a592db1a5fe3cf8752521987e1/crates/proc_macro_api/src/msg.rs#L31-L52

and call `std::env::set_env` when serving a request.

### PR DESCRIPTION
**Pull request by vlad20012**
Fixes #11079

### CODE REVIEW

File: crates/proc_macro_srv/src/lib.rs [hunk 1]
```diff
@@ -56,6 +66,11 @@ impl ProcMacroSrv {
                 None => env::remove_var(k),
             }
         }
+        if let Some(dir) = prev_working_dir {
+            if let Err(err) = std::env::set_current_dir(dir) {
+                eprint!("Failed to set the current working dir to {:?}. Error: {:?}", dir, err)
```
Comment (lnicola) [inline crates/proc_macro_srv/src/lib.rs:71]:
```suggestion
                eprint!("Failed to set the current working dir to {}. Error: {:?}", dir, err)
```

, I think.

Comment (vlad20012) [reply crates/proc_macro_srv/src/lib.rs:71]: PathBuf does not implement Display. I'm not sure how best to get around this

**Comment by lnicola** [comment]
What happens with `stderr` here? Do these end up in `rust-analyzer`'s `stderr`? Should they be `eprintln!`s?

File: crates/proc_macro_srv/src/lib.rs (hunk 1 above)
Comment (lnicola) [reply crates/proc_macro_srv/src/lib.rs]: `dir.display()`, I think

**Comment by vlad20012** [comment]
> What happens with stderr here? Do these end up in rust-analyzer's stderr (and in the Code output panel)? Should they be eprintln!s?

According to this line:

https://github.com/rust-analyzer/rust-analyzer/blob/e149a15edd354dd0f995532d16a3e8dc21367e06/crates/proc_macro_api/src/process.rs#L91

the expander inherits `stderr` from the main RA process.
Alternatively, we can:
1. Turn it into a hard error, i.e. consider a macro expansion as failed if we can't set the dir
2. Add the error to `Response::ExpandMacro` and log it somehow on the RA side

File: crates/proc_macro_srv/src/lib.rs [hunk 2]
```diff
@@ -56,6 +66,15 @@ impl ProcMacroSrv {
                 None => env::remove_var(k),
             }
         }
+        if let Some(dir) = prev_working_dir {
+            if let Err(err) = std::env::set_current_dir(&dir) {
+                eprintln!(
+                    "Failed to set the current working dir to {:?}. Error: {:?}",
```
Comment (lnicola) [inline crates/proc_macro_srv/src/lib.rs:72]:
```suggestion
                    "Failed to set the current working dir to {}. Error: {:?}",
```

**Comment by lnicola** [comment]
Yeah, I think it's fine.

bors d+

**Pull request merged**